src/safeposix/dispatcher.rs

```diff
@@ -550,6 +585,29 @@ pub extern "C" fn lindthreadremove(cageid: u64, pthreadid: u64) {
     cage.thread_table.remove(&pthreadid);
 }
 
+#[no_mangle]
+pub extern "C" fn lindgetsighandler(cageid: u64, signo: i32) -> u32 {
+    let cage = interface::cagetable_getref(cageid);
+    let pthreadid = interface::get_pthreadid();
+    let sigset = cage.sigset.get(&pthreadid).unwrap();
+    let pendingset = cage.sigset.get(&pthreadid).unwrap();
+
+    if !interface::lind_sigismember(sigset.load(interface::RustAtomicOrdering::Relaxed), signo) {
+        return match cage.signalhandler.get(&signo) {
+            Some(action_struct) => {
+                    action_struct.sa_handler // if we have a handler and its not blocked return it
+
+            },
+            None => 0, // if we dont have a handler return 0
+        };
+    } else { 
+        let mutpendingset = sigset.load(interface::RustAtomicOrdering::Relaxed);
```

Comment:
Do we have implicit locking on anything? If not, this load-then-store will always have a risk of race condition, isn't it?

Reply:
Similarly, latter in this PR, the pattern of "firstly load a sigset, after some modification on it, store it back" is prevalent, how is race condition prevented for those syscalls? Even though everything is atomic variable, we can still have situations like
syscall1 ( load X)
syscall2( load X)
syscall1(turn on some bits in X, store it back)
syscall2(turn on some bits in X, store it back)

It's just not obvious to me that, why situations like this will not happen.

Reply:
Yeah the cage.sigset is a DashMap so when we 'get' from it it explicitly locks the hashmap. Good thing to make sure about though. Thanks!